include/engine/edge_unpacker.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef EDGE_UNPACKER_H
+#define EDGE_UNPACKER_H
+
+#include "extractor/guidance/turn_instruction.hpp"
+#include "extractor/travel_mode.hpp"
+#include "engine/phantom_node.hpp"
+#include "osrm/coordinate.hpp"
+#include "util/guidance/turn_lanes.hpp"
+#include "util/typedefs.hpp"
+
+#include <vector>
+#include <stack>
+
+namespace osrm
+{
+namespace engine
+{
+
+/**
+ * Given a sequence of connected `NodeID`s in the CH graph, performs a depth-first unpacking of
+ * the shortcut
+ * edges.  For every "original" edge found, it calls the `callback` with the two NodeIDs for the
+ * edge, and the EdgeData
+ * for that edge.
+ *
+ * The primary purpose of this unpacking is to expand a path through the CH into the original
+ * route through the
+ * pre-contracted graph.
+ *
+ * Because of the depth-first-search, the `callback` will effectively be called in sequence for
+ * the original route
+ * from beginning to end.
+ *
+ * @param packed_path_begin iterator pointing to the start of the NodeID list
+ * @param packed_path_end iterator pointing to the end of the NodeID list
+ * @param callback void(const std::pair<NodeID, NodeID>, const EdgeData &) called for each
+ * original edge found.
+ */
+
+template <typename DataFacadeT, typename RandomIter, typename Callback>
+inline void UnpackCHEdge(DataFacadeT *facade,
```

Comment:
Name and parameters don't match up for me. Packed path indicates more than a single CH edge?